File: Mathematical/powerset/powerset.cpp
/* Program to generate powerset of given set */

#include<bits/stdc++.h>
using namespace std;

void powerset(char *arr){
	for(int i=0;i<pow(2,n);i++){
		for(int j=0;j<strlen(arr);j++){
			if(i & (1 << j))
				cout<<arr[j];
		}
		cout<<endl;
	}
}
int main(){
	char arr[100];
	gets(arr);
	powerset(arr);
	return 0;
}

File: Mathematical/powerset/powerset.py
# program to find the powerset
import math

def powerset(set,n):
	for i in range(0,(int)(math.pow(2,n))):
		for j in range(0,n):
			if (i & (1<<j)):
				print(set[j],end="")
		print("")

set = input().split(",")
powerset(set,len(set))
